### PR TITLE
fix the types for procedure-rename and procedure->method

### LINKED ISSUE #1033: How to use `procedure-rename` to rename a procedure?
### What version of Racket are you using?
7.9[cs]

### What program did you run?
```
(define add (procedure-rename + 'add))
(displayln (add 1 2 3))
```

### What should have happened?
`6`

### If you got an error message, please include it here.
```
Type Checker: cannot apply a function with unknown arity;
 function `add' has type Procedure which cannot be applied
  in: (add 1 2 3)
```

### PR DESCRIPTION
**Pull request by capfredf**
close #1033

### CODE REVIEW

**Comment by sorawee** [comment]
I guess it's not possible to fix `procedure-reduce-arity` in the same way, right?

**Comment by samth** [comment]
That definitely needs a special case in the type checker, rather than just a type.

File: typed-racket-lib/typed-racket/base-env/base-env.rkt [hunk 1]
```diff
@@ -1179,8 +1179,8 @@
 [procedure? (make-pred-ty top-func)]
 [compose (-poly (a b c) (-> (-> b c) (-> a b) (-> a c)))]
 [compose1 (-poly (a b c) (-> (-> b c) (-> a b) (-> a c)))]
-[procedure-rename (-> top-func -Symbol top-func)]
-[procedure->method (-> top-func top-func)]
+[procedure-rename (-poly (a) (-> (-Inter top-func a) -Symbol (-Inter top-func a)))]
```
Comment (samth) [inline typed-racket-lib/typed-racket/base-env/base-env.rkt:1182]: Why do we need the intersection on the result type?

Comment (capfredf) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt:1182]: Actually, we don't. I'm pushing a fix on that.

**Review by samth — APPROVED** [review]

**Pull request merged**